frame/salary/src/lib.rs

```diff
@@ -0,0 +1,512 @@
+		/// `RegistrationPeriod`.
+		#[pallet::constant]
+		type PayoutPeriod: Get<Self::BlockNumber>;
+
+		/// The total budget per cycle.
+		///
+		/// This may change over the course of a cycle without any problem.
+		#[pallet::constant]
+		type Budget: Get<BalanceOf<Self, I>>;
+	}
+
+	pub type CycleIndexOf<T> = <T as frame_system::Config>::BlockNumber;
+	pub type BalanceOf<T, I> = <<T as Config<I>>::Paymaster as Pay>::Balance;
+	pub type IdOf<T, I> = <<T as Config<I>>::Paymaster as Pay>::Id;
+	pub type StatusOf<T, I> =
+		StatusType<CycleIndexOf<T>, <T as frame_system::Config>::BlockNumber, BalanceOf<T, I>>;
+	pub type ClaimantStatusOf<T, I> = ClaimantStatus<CycleIndexOf<T>, BalanceOf<T, I>, IdOf<T, I>>;
+
+	/// The overall status of the system.
+	#[pallet::storage]
+	pub(super) type Status<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, StatusOf<T, I>, OptionQuery>;
+
+	/// The status of a claimant.
+	#[pallet::storage]
+	pub(super) type Claimant<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Twox64Concat, T::AccountId, ClaimantStatusOf<T, I>, OptionQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config<I>, I: 'static = ()> {
+		/// A member is inducted into the payroll.
+		Inducted { who: T::AccountId },
+		/// The next cycle begins.
+		Registered { who: T::AccountId, amount: BalanceOf<T, I> },
+		/// A payment happened.
+		Paid {
+			who: T::AccountId,
+			beneficiary: T::AccountId,
+			amount: BalanceOf<T, I>,
+			id: <T::Paymaster as Pay>::Id,
+		},
+		/// The next cycle begins.
+		CycleStarted { index: CycleIndexOf<T> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T, I = ()> {
+		/// The salary system has already been started.
+		AlreadyStarted,
+		/// The account is not a ranked member.
+		NotMember,
+		/// The account is already inducted.
+		AlreadyInducted,
+		// The account is not yet inducted into the system.
+		NotInducted,
+		/// The member does not have a current valid claim.
+		NoClaim,
+		/// The member's claim is zero.
+		ClaimZero,
+		/// Current cycle's registration period is over.
+		TooLate,
+		/// Current cycle's payment period is not yet begun.
+		TooEarly,
+		/// Cycle is not yet over.
+		NotYet,
+		/// The payout cycles have not yet started.
+		NotStarted,
+		/// There is no budget left for the payout.
+		Bankrupt,
+		/// There was some issue with the mechanism of payment.
+		PayError,
+		/// The payment has neither failed nor succeeded yet.
+		Inconclusive,
+		/// The cycle is after that in which the payment was made.
+		NotCurrent,
+	}
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Start the first payout cycle.
+		///
+		/// - `origin`: A `Signed` origin of an account which is a member of `Members`.
+		#[pallet::weight(T::WeightInfo::init())]
+		#[pallet::call_index(0)]
+		pub fn init(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
```

Comment:
It looks like anyone can call this and does not need to be a member of `Members`. Are the docs or implementation the intended behavior?

Reply:
Will update docs.